code/game/objects/explosion.dm

```diff
@@ -123,9 +123,23 @@
 				if(flame_dist && prob(40) && !istype(T, /turf/space) && !T.density)
 					new /obj/effect/hotspot(T) //Mostly for ambience!
 				if(dist > 0)
-					for(var/atom_movable in T.contents)	//bypass type checking since only atom/movable can be contained by turfs anyway
-						var/atom/movable/AM = atom_movable
-						if(AM && AM.simulated)	AM.ex_act(dist)
+					if(istype(T, /turf/simulated))
+						var/turf/simulated/S = T
+						var/affecting_level
+						if(dist == 1)
+							affecting_level = 1
+						else
+							affecting_level = S.is_shielded() ? 2 : (S.intact ? 2 : 1)
+						for(var/atom_movable in S.contents)	//bypass type checking since only atom/movable can be contained by turfs anyway

```

Comment:
I know it's not yours, I dug and dug through your repo's history but got lazy and couldn't trace it.
As for what else goes in a contents list, anything with a loc, so ```/atom``` downwards NOT ```/atom/movable``` downwards.
Granted nothing in these loops appears to require anything ```/atom/movable``` specific, but it doesn't change the fact those typecasts are erroneous.